### ISSUE TITLE
[BUG] spliceRows doesnt move formulas of following rows

### ISSUE THREAD

**Issue by MR4online** [exceljs/exceljs#1188]
## 🐛 Bug Report

When I use spliceRows() to add more rows inside a table, all formulas in cells/rows after the added row are NOT moved.
Text, Style, ... of the Cells are moved, Formulas NOT!

EDIT:
It looks like this happens only for rows further down. May be, when empty rows are between the inserted and the cell with formula.

Lib version: 3.8.1

## Steps To Reproduce

Have an Excel File with a row "2" and e.g. a SUM(D2:D5) in D6.
Now use spliceRows(2, [something]).
The formula stays in D6.

## The expected behaviour:

The Formular moves to D7, because I insered a row before.

**Comment by MR4online** [comment]
## Possible Solution

Looks like i can fix this by overwriting the cell value in:
worksheet.js line 404

```
// org code
          rSrc.eachCell({includeEmpty: true}, (cell, colNumber) => {
            rDst.getCell(colNumber).style = cell.style;
// added code
            if (rDst.getCell(colNumber).value && rDst.getCell(colNumber).value.formula) {
              rDst.getCell(colNumber).value = {
                formula: rDst.getCell(colNumber).value.formula
              };
            }
          });
```

because the problem seems to be, that cell.value lookes like:

```
{
formula: 'SUM((INDIRECT(ADDRESS(ROW()-2,COLUMN()))+INDIRECT(ADDRESS(ROW()-1,COLUMN())))/2)',
ref: 'D6',
shareType: 'array'
}
```

whyever this results in the fomula to be placed in D6, not the actual cell position D7.

_Testet with Microsoft Excel for Mac 16.35 (current version)._

**Comment by MR4online** [comment]
@Siemienik may you please have a look, if this makes sense? Thanks a lot!

**Comment by Siemienik** [comment]
as metion in readme:
```javascript
// remove one row and insert two more.
// Note: rows 4 and below will be shifted down by 1 row.
var newRow3Values = [1,2,3,4,5];
var newRow4Values = ['one', 'two', 'three', 'four', 'five'];
worksheet.spliceRows(3, 1, newRow3Values, newRow4Values);
```
you should do `spliceRows(2, 0, [something])` not `spliceRows(2, [something])`

**Comment by MR4online** [comment]
> as metion in readme:
> 
> ```js
> // remove one row and insert two more.
> // Note: rows 4 and below will be shifted down by 1 row.
> var newRow3Values = [1,2,3,4,5];
> var newRow4Values = ['one', 'two', 'three', 'four', 'five'];
> worksheet.spliceRows(3, 1, newRow3Values, newRow4Values);
> ```
> 
> you should do `spliceRows(2, 0, [something])` not `spliceRows(2, [something])`

thats what i already do!

```
workbook
.getWorksheet(WORKSHEET_ID)
.spliceRows(ROW_ID, 0, ["newLine"]);
```

**Comment by sdg9670** [comment]
Note and auto filters are also not deleted.

**Comment by MR4online** [comment]
looks like its fixed in 4.2.0 (or may be befor).
